core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/resolver/ScoreResolver.java

```diff
@@ -43,7 +43,7 @@
      * @param defaultScoreResolver the default score resolver to look up scores for nested operations
      * @return the score for the operation, otherwise null if not found
      */
-    default Integer getScore(final T operation, final ScoreResolver defaultScoreResolver) {
+    default Integer getScore(final T operation, final ScoreResolver<T> defaultScoreResolver) {
```

Comment:
I don't think you should change this interface method signature, as it will cause breaking changes.